### PR TITLE
bugfix for FY5253 case with bunched yearends

### LINKED ISSUE #14774: AssertionError in FY5253.rollback()
Hi,

I'm trying to do North American (Year,WeekOfYear) <-> Period (W-SAT) conversions, for calculations involving fiscal year week numbers, etc. From what I understand pandas.offsets.FY5253 can be used for this, but I'm getting a weird assertion error than should probably not happen.

#### Code

```python
import pandas as pd
fy = pd.offsets.FY5253(n=1, weekday=5, startingMonth=12, variation='nearest')
fy.rollback(pd.Timestamp('2004-01-01'))
```

#### Expected Output

I think the output should be equal to the one you get with
```python
fy.rollback(pd.Timestamp('2003-12-31'))
```
which is
```python
Timestamp('2002-12-28 00:00:00')
```

#### Output of ``pd.show_versions()``

<details>

commit: None
python: 2.7.12.final.0
python-bits: 64
OS: Linux
OS-release: 4.2.0-35-generic
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.19.1
nose: 1.3.7
pip: 8.1.2
setuptools: 28.7.1
Cython: 0.23.4
numpy: 1.11.2
scipy: 0.17.1
statsmodels: 0.6.1
xarray: None
IPython: 4.1.2
sphinx: 1.3.5
patsy: 0.4.0
dateutil: 2.6.0
pytz: 2016.7
blosc: None
bottleneck: 1.0.0
tables: 3.2.2
numexpr: 2.5
matplotlib: 1.5.1
openpyxl: 2.3.2
xlrd: 0.9.4
xlwt: 1.0.0
xlsxwriter: 0.8.4
lxml: 3.6.0
bs4: 4.4.1
html5lib: None
httplib2: 0.9.2
apiclient: 1.4.2
sqlalchemy: 1.0.12
pymysql: None
psycopg2: 2.6.1 (dt dec pq3 ext)
jinja2: 2.7.3
boto: 2.38.0
pandas_datareader: None

</details>

### PR DESCRIPTION
**Pull request by jbrockmendel**
Also achieves coverage for one of the uncovered branches in `FY5253.apply`

- [x] closes #14774
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Comment by jreback** [comment]
cc @cancan101

**Comment by jreback** [comment]
release note pls.

**Comment by jbrockmendel** [comment]
Ping.

File: pandas/tests/tseries/offsets/test_fiscal.py [hunk 1]
```diff
@@ -604,3 +605,17 @@ def test_offset(self):
...
                             datetime(2013, 3, 30))
+
+
+def test_bunched_yearends():
+    # GH#14774 cases with two fiscal year-ends in the same calendar-year
+    fy = FY5253(n=1, weekday=5, startingMonth=12, variation='nearest')
+    dt = Timestamp('2004-01-01')
+    assert fy.rollback(dt) == Timestamp('2002-12-28')
+    assert (-fy).apply(dt) == Timestamp('2002-12-28')
+    assert dt - fy == Timestamp('2002-12-28')
```
Comment (jreback) [inline pandas/tests/tseries/offsets/test_fiscal.py]: can you add from the OP: ``fy.rollback(pd.Timestamp('2003-12-31'))`` (which I think is the same result) as well

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
cc @ronif can you have a look.

**Comment by jbrockmendel** [comment]
Just rebased and pushed; no new changes.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks

**Pull request merged**